server/controllers/joinLeaveCancelController.js

```diff
@@ -0,0 +1,107 @@
+"use strict";
+
+const dbName = "ParkGames";
+const collectionEvents = "Events";
+const collectionParticipants = "Participants";
+const collectionUserEvents = "UserEvents";
+const collectionRooms = "Rooms";
+const assert = require("assert");
+var ObjectId = require("mongodb").ObjectID;
+//env vairables
+require("dotenv").config();
+const {
+  //handleJoinEvent.
+  getEventById,
+  getParticipantsById,
+  getMatchingParticipant,
+  addParticipant,
+  addUserEvent,
+  removeParticipant,
+  removeUserEvent,
+  //handleLeaveEvent.
+
+  //handleCancelEvent.
+} = require("../services/joinLeaveCancelService");
+
+//@endpoint POST /joinEvent
+//@desc join the event selected from ViewActivity component.
+//@access PRIVATE - will need to validate token? YES
+const handleJoinEvent = async (req, res, next) => {
+  const participantDetails = req.body.participantDetails;
```

Comment:
This should be inside the `try/catch`, if for some reason `req.body` doesn't exist nothing will handle this error. It's unlikely, but as a rule we want every piece of code inside our top level `try/catch`

Reply:
Also we need more validation here. We have `participantDetails` and `eventInformation` but what happens if we don't have `eventInformation._id`, `participantDetails.userId`, etc?